Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/API/UI/RideSummary.hs

```diff
@@ -0,0 +1,41 @@
+{-
+ Copyright 2022-23, Juspay India Pvt Ltd
+
+ This program is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General Public License
+
+ as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version. This program
+
+ is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+
+ or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more details. You should have received a copy of
+
+ the GNU Affero General Public License along with this program. If not, see <https://www.gnu.org/licenses/>.
+-}
+
+module API.UI.RideSummary where
+
+import Data.Time (Day)
+import Domain.Action.UI.RideSummary as RDS
+import qualified Domain.Types.Merchant as Merchant
+import qualified Domain.Types.Person as DP
+import Environment
+import Kernel.Prelude
+import Kernel.Types.Id
+import Kernel.Utils.Common
+import Servant
+import Tools.Auth
+
+type API =
+  "rideSummary"
+    :> ( "list"
+           :> TokenAuth
+           :> MandatoryQueryParam "fromDate" Day
+           :> MandatoryQueryParam "toDate" Day
```

Comment:
Maybe not have these mandatory and send today's stats by default?

Reply:
Actually the use case here is not like that 
We will be sending fromDate and toDate from frontend (always Monday to sunday), it doesn't matter if today is tuesday or wednesday.
you can see this from figma
https://www.figma.com/file/4zT4QhImxqphGf56DpAjcV/Beckn-Release-Version?type=design&node-id=82227-37499&mode=design&t=FPCQXmpWAatIQVmP-0

Reply:
In that case, why even take these as params, just send current week's data? Anyways, its fine. You can send weekly data by default and make these optional